### DISCUSSION TITLE
Trouble with relations that already contain a relation..

### DISCUSSION THREAD

**Discussion by cpt-n3mo** [directus/directus#12138] [Archive]
Good day..

Im running in some trouble and im not sure how to solve it..

when i create the following collections:
```
social_media < social_media_child 
address < address_child
footer
```
when i now do a relation to either social_media or address from footer collection, every thing just stops working as expected. i cant select them with a many to any relation, when i use a many to many, and add a existing item it gives a list that is empty in values but a incorrect amount of rows is shown.

im i doing something wrong here..?  am i not able to use this it in this way ?

kind regards

**Comment by rijkvanzanten** [comment]
What is the configuration for those relationships? Could be that you're using the wrong fields for the foreign keys, but it's hard to tell from the provided info 🙂

**Comment by cpt-n3mo** [comment]
i see, i was thinking it would be something like that. this is the config of the relations

social_media_child contains:
```
ID
href: String
media_type: dropdown
```

social_media contains
```
ID
social_media: many to many < social_media_child 
```

adress hes the same kind of structure..

edit:
when i try to remove the dorked relations i also get a error, so i cant remove the item nor the collection.
while trying to remove footer collection when it contains dorked relations.
```
{
  "errors": [
    {
      "message": "drop table `footer` - ER_ROW_IS_REFERENCED_2: Cannot delete or update a parent row: a foreign key constraint fails",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```

this error is given when i want to remove a dorked relation.
```
{
  "errors": [
    {
      "message": "alter table `footer` drop `adress` - ER_DROP_INDEX_FK: Cannot drop index 'footer_adress_foreign': needed in a foreign key constraint",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```


kind regards

**Reply by rijkvanzanten** [reply]
What are the fields on the junction collection from social_media -> ?? -> social_media_child ?
